core/utils.py

```diff
@@ -1278,3 +1278,10 @@ def url_open(source_url: str) -> str:
     # The type ignore is needed, because typestubs define the return type
     # of 'urlopen' as 'Any' which is wrong.
     return urllib.request.urlopen(source_url, context=context) # type: ignore[no-any-return]
+
+
+def generate_random_str_from_regex() -> str:
+    """Generate random string for a particular regex pattern."""
```

Comment:
Where is that reges pattern? Also, this is missing a Returns section in the docstring. Also, if it is only used in tests, then should it be moved to _test_utils.py_?

Reply:
yup, I added it in test_utils. please have a look